<a href="https://colab.research.google.com/github/AlifHammam/data-science-2026/blob/main/Pertemuan2_AlifHamammamMultazam_240401010043.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Pertemuan 2 - Struktur Data Python, NumPy & Pandas

**Nama Lengkap:** Alif Hamammam Multazam  
**NIM:** 240401010043  
**Kelas:** IF403

## 1. Struktur Data Python

In [14]:
# List Operations
data = [10, 20, 'Python', True, 3.14]

print(data[0])    # akses index pertama
print(data[-1])   # akses dari belakang
print(data[1:3])  # slicing

data.append(99)
data.insert(1, 55)
data.remove(20)
print(data)

10
3.14
[20, 'Python']
[10, 55, 'Python', True, 3.14, 99]


In [15]:
# Dictionary Operations
mahasiswa = {
    'nama': 'Alif Hamammam Multazam',
    'nim': '240401010043',
    'prodi': 'Informatika',
    'ipk': 3.96
}

print(mahasiswa['nama'])
print(mahasiswa.get('usia', 0))  # default jika key tidak ada

mahasiswa['angkatan'] = 2024
del mahasiswa['nim']
print(mahasiswa)

Alif Hamammam Multazam
0
{'nama': 'Alif Hamammam Multazam', 'prodi': 'Informatika', 'ipk': 3.96, 'angkatan': 2024}


## 2. NumPy: Komputasi Numerik

In [16]:
import numpy as np

# Membuat array
a1D = np.array([1, 2, 3, 4, 5])
a2D = np.array([[1,2,3],[4,5,6]])

zeros = np.zeros((3, 4))
ones  = np.ones((2, 3))
eye   = np.eye(4)

arange = np.arange(0, 20, 2)
linsp  = np.linspace(0, 1, 11)

rng      = np.random.default_rng(seed=42)
rand_int = rng.integers(0, 100, (3, 4))
rand_flt = rng.normal(loc=0, scale=1, size=(100,))

print('a1D:', a1D)
print('a2D shape:', a2D.shape)
print('arange:', arange)
print('linspace:', linsp)

a1D: [1 2 3 4 5]
a2D shape: (2, 3)
arange: [ 0  2  4  6  8 10 12 14 16 18]
linspace: [0.  0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1. ]


In [17]:
# Vektorisasi dan Fungsi Statistik
a = np.array([1, 2, 3, 4, 5])
b = np.array([10, 20, 30, 40, 50])

print(a + b)
print(a * b)
print(a ** 2)

data = rng.normal(0, 1, 1000)
print(f'Mean : {np.mean(data):.4f}')
print(f'Std  : {np.std(data):.4f}')
print(f'Min  : {np.min(data):.4f}')
print(f'Max  : {np.max(data):.4f}')
print(f'Q1   : {np.percentile(data, 25):.4f}')
print(f'Q3   : {np.percentile(data, 75):.4f}')

[11 22 33 44 55]
[ 10  40  90 160 250]
[ 1  4  9 16 25]
Mean : -0.0163
Std  : 1.0008
Min  : -3.6484
Max  : 3.1789
Q1   : -0.6897
Q3   : 0.6163


## 3. Pandas: Manipulasi DataFrame

In [18]:
import pandas as pd

# Pandas Series
nilai = pd.Series(
    [85, 92, 78, 95],
    index=['Alice', 'Bob', 'Caca', 'Dina']
)

print(nilai['Bob'])
print(nilai[nilai > 80])
print(nilai.mean())
print(nilai.describe())

92
Alice    85
Bob      92
Dina     95
dtype: int64
87.5
count     4.000000
mean     87.500000
std       7.593857
min      78.000000
25%      83.250000
50%      88.500000
75%      92.750000
max      95.000000
dtype: float64


In [19]:
# Pandas DataFrame
data = {
    'nama': ['Ali', 'Budi', 'Caca', 'Dina'],
    'usia': [22, 25, 21, 23],
    'kota': ['Jakarta', 'Bandung', 'Surabaya', 'Medan'],
    'ipk' : [3.8, 3.5, 3.9, 3.7]
}
df = pd.DataFrame(data)

print(df.shape)
print(df.dtypes)
print(df)

(4, 4)
nama     object
usia      int64
kota     object
ipk     float64
dtype: object
   nama  usia      kota  ipk
0   Ali    22   Jakarta  3.8
1  Budi    25   Bandung  3.5
2  Caca    21  Surabaya  3.9
3  Dina    23     Medan  3.7


## 4. Pertanyaan Latihan — Dataset Titanic

In [20]:
# Load dataset Titanic
titanic_url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(titanic_url)
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


### Q1 — Jumlah baris, kolom, dan nama kolom

In [21]:
print('Shape:', df.shape)  # (baris, kolom)
print('Kolom:', df.columns.tolist())

Shape: (891, 12)
Kolom: ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


### Q2 — Missing values per kolom

In [22]:
missing = df.isnull().sum()
missing_pct = (df.isnull().sum() / len(df) * 100).round(2)

missing_df = pd.DataFrame({'Jumlah Missing': missing, 'Proporsi (%)': missing_pct})
print(missing_df[missing_df['Jumlah Missing'] > 0])

          Jumlah Missing  Proporsi (%)
Age                  177         19.87
Cabin                687         77.10
Embarked               2          0.22


### Q3 — Persentase penumpang yang selamat

In [23]:
pct_selamat = np.mean(df['Survived']) * 100
print(f'Persentase selamat: {pct_selamat:.2f}%')

Persentase selamat: 38.38%


### Q4 — Filter penumpang wanita kelas 1

In [24]:
wanita_k1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]
print('Jumlah:', len(wanita_k1))
print(f'Rata-rata usia: {wanita_k1["Age"].mean():.2f} tahun')

Jumlah: 94
Rata-rata usia: 34.61 tahun


### Q5 — Tingkat keselamatan per kelas penumpang

In [25]:
survival_rate = df.groupby('Pclass')['Survived'].mean().round(3)
print(survival_rate)
print(f'\nKelas dengan tingkat keselamatan tertinggi: Kelas {survival_rate.idxmax()}')

Pclass
1    0.630
2    0.473
3    0.242
Name: Survived, dtype: float64

Kelas dengan tingkat keselamatan tertinggi: Kelas 1


### Q6 — Analisis tambahan: Tingkat keselamatan berdasarkan jenis kelamin

In [26]:
# Saya tertarik melihat perbedaan tingkat selamat antara laki-laki dan perempuan
survival_sex = df.groupby('Sex')['Survived'].mean().round(3)
print(survival_sex)

# Breakdown lebih detail: kelas + jenis kelamin
detail = df.groupby(['Pclass', 'Sex'])['Survived'].mean().unstack().round(3)
print('\nDetail per kelas dan jenis kelamin:')
print(detail)

Sex
female    0.742
male      0.189
Name: Survived, dtype: float64

Detail per kelas dan jenis kelamin:
Sex     female   male
Pclass               
1        0.968  0.369
2        0.921  0.157
3        0.500  0.135


## Kesimpulan

Pada praktikum pertemuan kedua ini saya mempelajari struktur data Python (list, tuple, dictionary, set), komputasi numerik menggunakan NumPy, serta manipulasi data tabel menggunakan Pandas.

Melalui latihan dengan dataset Titanic, saya memahami cara memuat data dari URL, mengeksplorasi struktur dataset, menangani missing values, melakukan filtering dengan boolean indexing, serta mengelompokkan data menggunakan groupby. Dari analisis yang dilakukan, ditemukan bahwa sekitar 38% penumpang selamat, dengan penumpang wanita dan kelas 1 memiliki tingkat keselamatan yang jauh lebih tinggi dibanding yang lainnya.

Keterbatasan pada praktikum ini adalah analisis masih bersifat deskriptif dan belum masuk ke tahap visualisasi maupun pemodelan prediktif. Namun, praktikum ini sangat membantu dalam memahami cara kerja Pandas dan NumPy sebagai fondasi utama sebelum masuk ke materi Data Science yang lebih kompleks.